keymanapp/keyman
# fix(core): Fix nuances when updating context 🌱

Fixes #10100.

# User Testing

## Preparations

- The tests should be run on Linux, Windows and Mac.
- Install build artifacts of this PR. [(Instructions for Linux)](https://github.com/keymanapp/keyman/wiki/How-to-test-artifacts-from-pull-requests-for-Keyman-for-Linux)
- on Linux install the .deb version of Firefox and/or Chrome/Chromium and remove the snap version
- Reboot

## Tests

- GROUP_WIN: Windows
- GROUP_MAC: macOS
- GROUP_LINUX: Ubuntu 22.04 with X11

Open https://keyman.com/keyboards in a browser and run the following tests:

   * TEST_KHMER_ANGKOR: Khmer Angkor
      * Ensure the key combination <kbd>AltGr</kbd> <kbd>B</kbd> produces `ឞ`. Note that on macOS, the function key to use is right `option`.
      * Type the following (US) keys in sequence: <kbd>s</kbd> <kbd>u</kbd> <kbd>j</kbd> <kbd>r</kbd> <kbd>k</kbd>. You should see `ស្រុក`.
   * TEST_SIL_EURO_LATIN: SIL Euro Latin
      * Type <kbd>%</kbd> <kbd>a</kbd>. Ensure the output is `ǎ`
   * TEST_GALAXIE_HEBREW: Galaxie Hebrew or another good SIL RTL keyboard
      * Type <kbd>b</kbd> <kbd>\\</kbd> <kbd>=</kbd> <kbd>r</kbd> <kbd>@</kbd> <kbd>a</kbd> <kbd>v</kbd> <kbd>!</kbd> <kbd>y</kbd> <kbd>t</kbd> (note the third-last key is the exclamation mark!) to output `בְּרֵאשִׁית`. FAILED if the output is different.
   * TEST_SIL_IPA: sil_ipa
      * Ensure the key sequence <kbd>n</kbd> <kbd>></kbd> produces `ŋ`.
   * TEST_CJK_IMX: Korean KORDA Jamo (SIL)
      * Type <kbd>h</kbd> <kbd>a</kbd> <kbd>n</kbd> `<space>` <kbd>g</kbd> <kbd>e</kbd> <kbd>u</kbd> <kbd>l</kbd> `<space>`. Verify that the result is `한글`.
